### PR TITLE
Bug 1033686 - Pre-search and post-search are visible at the same time 

### PR DESCRIPTION
**Pull request by EricEdens**
Much of the changes are renaming; the substantial changes are:
-     MainActivity.java
-     MainActivityTest.java

### CODE REVIEW

**Comment by ncalexan** [comment]
OK, the code looks mostly fine, but it's hard to follow your sequence of git commits.  A few general comments:
- Things that touch gradle and geckoView should get committed whenever you need them; they don't touch m-c so in general, just get that stuff out there.
- Your test code should live in `app/src/test` not in `app/src/main`.
- Prefer rebase rather than merge when you're developing locally.  I'd like to see this whole sequence squashed to maybe 3 or 4 commits: renaming, fixes, tests.
- The m-c patch you generate from this will need to be based off my #18

**Comment by EricEdens** [comment]
Thanks Nick!

I made the updates, and I'll pull #18 momentarily. I think Gradle looks for "androidTest", so I used that. http://tools.android.com/tech-docs/new-build-system/user-guide#TOC-Project-Structure

**Comment by ncalexan** [comment]
I have some questions about the activity and fragment lifecycle.  We might just want to take 10 minutes on Vidyo tomorrow; let me know.

The git commits are much better!  (**Much** easier to read.)  In general, for your commit messages, we like:

```
Bug XXX - Part YYY: Explain what your patch does.

Add details about your patch here, if necessary.
```

So: `Bug 1033686 - Part 2: Rename Auto->Search, Card->Pre, Detail->Post.`

**Comment by ncalexan** [comment]
For the test commit, I'm not sure we really want to do this at all, because it will be clear during use if we're busted.  But: http://stackoverflow.com/questions/9323279/how-to-test-if-a-fragment-view-is-visible-to-the-user seems relevant.  I think this is tricky in general.  Perhaps better to test that our activity's internal state is correct, or our search activity model reflects the actions we expect.

**Comment by EricEdens** [comment]
This is a WIP update. As an aside, I rebased and did a force update, and it looks like all of your review comments were lost :/

Instead of managing fragments with the fragment manager, this is using the layout xml file. The net effect is the same, but it removes a lot of the manual lifecycle management that we talked about last night.

**Comment by ncalexan** [comment]
Further to discussion with @lucasr, let's land this.  (Here and on central.)  @ericedens, in android-sync we land with a `--non-ff` merge; we make the merge commit message exactly the same as the message that we'll land into m-c, including the `r=nalexander` tag.  Might as well do the same here.
